presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayContains.java

```diff
@@ -87,6 +87,10 @@ public static Boolean contains(Type type, Slice slice, Slice value)
         for (int i = 0; i < arrayBlock.getPositionCount(); i++) {
             if (arrayBlock.isNull(i)) {
                 foundNull = true;
+                if (valueBlock.isNull(0)) {
```

Comment:
This if is unnecessary because this function is not nullable (see last argument on line 79). Therefore, it will never evaluate to true.

For `contains ([null], null)`, the expected return is `null`. I checked just now and this case is not covered in test.

This comments applies to other changes as well.